src/main/java/kr/co/vcnc/haeinsa/HaeinsaDelete.java

```diff
@@ -107,6 +108,22 @@ public void add(HaeinsaMutation newMutation) {
         }
     }
 
+    FilterResult filter(HaeinsaPut put) {
+        HaeinsaDeleteTracker deleteTracker = new HaeinsaDeleteTracker();
+        deleteTracker.add(this, 0);
+        FilterResult filterResult = new FilterResult(row);
+        for (Entry<byte[], NavigableSet<HaeinsaKeyValue>> entry : put.getFamilyMap().entrySet()) {
+            for (HaeinsaKeyValue keyValue : entry.getValue()) {
+                if (deleteTracker.isDeleted(keyValue, 1)) {
```

Comment:
overloading `isDeleted(keyValue)` would be make it readable (just call `isDeleted(keyValue, Long.MAX_VALUE)` inside)